File: Chapter 05/Python/section_5_4_server_monitor/after/tests/server_monitor_test.py
from ..core.domain.server_monitor import ServerMonitor
from ..tests.fake_alert_port import FakeAlertPort

class ServerMonitorTests:
    """
    ARCHITECTURAL TEST
    Fulfills the Scribe role by proving the test passes without hitting a real API.
    """

    @staticmethod
    def run() -> None:
        print("--- RUNNING ARCHITECTURAL TEST: HEXAGONAL ---")
        
        # Arrange
        fake_port = FakeAlertPort()
        monitor = ServerMonitor(fake_port)

        # Act
        print("Test Action: Checking temperature at 96 degrees...")
        monitor.check_temperature(96)

        # Assert
        if len(fake_port.sent_messages) == 1 and "Take cover" in fake_port.sent_messages[0]:
            print("SUCCESS: Alert sent correctly to the Port.")
        else:
            print("FAIL: Alert logic failed verification.")

File: Chapter 05/JS/section_5_4_server_monitor/after/tests/serverMonitorTest.js
const ServerMonitor = require('../core/domain/serverMonitor');
const FakeAlertPort = require('./fakeAlertPort');

/**
 * ARCHITECTURAL TEST.
 * Proves that by using Ports, we can verify business logic in total isolation.
 */
class ServerMonitorTests {
    /**
     * Executes the hexagonal test suite.
     */
    static run() {
        console.log("--- RUNNING ARCHITECTURAL TEST: HEXAGONAL ---");
        
        // 1. Arrange: Setup the "Fake" environment
        const fakePort = new FakeAlertPort();
        const monitor = new ServerMonitor(fakePort);

        // 2. Act: Trigger the logic
        console.log("Test Action: Checking temperature at 96 degrees...");
        monitor.checkTemperature(96);

        // 3. Assert: Verify the Scribe recorded the correct interaction
        if (fakePort.sentMessages.length === 1 && fakePort.sentMessages[0].includes("Take cover")) {
            console.log("SUCCESS: Alert sent correctly to the Port.");
        } else {
            console.error("FAIL: Alert logic failed verification.");
        }
    }
}

module.exports = ServerMonitorTests;